### PR TITLE
Update Gemmini schedules

### PR DESCRIPTION
**Pull request by yamaguchi1024**
Generalize 512x512x512 Gemmini schedules to handle 256 multiples of N and M, which was necessary for producing the [performance results](https://docs.google.com/spreadsheets/d/1oNiDLUAn_DfDARSMfhJ72V3mUMZxT_oNVVYPGH07gtM/edit?usp=sharing)

### CODE REVIEW

File: tests/pldi24/test_gemmini_matmul_ae_rewrite.py [hunk 1]
```diff
@@ -102,9 +104,9 @@ def test_matmul_ae(golden):
     res_alloc = res_load.prev()
 
     # Tile loops for a scratchpad and an accumulator
-    gemmini, _ = tile_loops_top_down(gemmini, [(i_loop, 16), (j_loop, 16)])
-    gemmini, [_, j_outer] = tile_loops_top_down(gemmini, [(i_loop, 16), (j_loop, 16)])
-    gemmini, _ = tile_loops_top_down(gemmini, [(k_loop, 16)])
+    gemmini, _ = perfect_tile(gemmini, [(i_loop, 16), (j_loop, 16)])
```
Comment (skeqiqevian) [inline tests/pldi24/test_gemmini_matmul_ae_rewrite.py:107]: Previously, `tile_loops_top_down` worked for both `conv` and `matmul` Gemmini schedules. Does `perfect_tile` not apply to the `conv` schedule?

**Review by skeqiqevian — APPROVED** [review]
Schedule-wise LGTM. As far as I can tell, in addition to generalizing M and N to any multiple of 256, you were also to reduce the accumlator's accumulation size. It wasn't clear to me what schedule changes enabled that optimization, so an explanation would be helpful!

File: tests/pldi24/gemmini_schedules.py [hunk 2]
```diff
@@ -139,10 +139,10 @@ def add_guard(p, c):
         if not isinstance(c, pc.ForCursor):
             break
         try:
-            hi = c.hi().value()
+            hi = expr_to_string(c.hi())
             name = c.name()
             child = p.forward(c).body()[0]
-            p = remove_loop(p, c)
+            p = remove_loop(p, c, unsafe_disable_check=True)
```
[the remark this answers is not in the document]
Comment (yamaguchi1024) [reply tests/pldi24/gemmini_schedules.py:145]: #652 is the issue about the remove_loop failure

File: tests/pldi24/test_gemmini_matmul_ae_rewrite.py (hunk 1 above)
Comment (yamaguchi1024) [reply tests/pldi24/test_gemmini_matmul_ae_rewrite.py:107]: `perfect_tile` assumes that the loop is perfectly divisible even when the loop bound is not constant, but this makes me realize that we can just `try/except` that in `tile_loops_top_down` too. Maybe I'll change that

**Pull request merged**